Pr/fix fullscreen

uli42 opened the pull request.

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Window.c:
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
 
+    /*
+     * FIXME: These are 0 most of the time nowadays. The effect is,
+     * that the window is moving a bit to right/bottom everytime
```
s/everytime/every time/

sunweaver replied:
https://www.grammarly.com/blog/everytime-every-time/

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Window.c:
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
+    /*
+     * FIXME: These are 0 most of the time nowadays. The effect is,
+     * that the window is moving a bit to right/bottom everytime
+     * Fullscreen is left. To fix this ask the window manager the
```
s/Fullscreen/fullscreen/

sunweaver replied:
s/Fullscreen/fullscreen/

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Window.c:
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
+    /*
+     * FIXME: These are 0 most of the time nowadays. The effect is,
+     * that the window is moving a bit to right/bottom everytime
+     * Fullscreen is left. To fix this ask the window manager the
```
s/ask the window manager the frame extents/query the frame extents from the window manager/

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Window.c:
```diff
@@ -1003,6 +1008,13 @@ void nxagentSwitchAllScreens(ScreenPtr pScreen, Bool switchOn)
+     * FIXME: These are 0 most of the time nowadays. The effect is,
+     * that the window is moving a bit to right/bottom everytime
+     * Fullscreen is left. To fix this ask the window manager the
+     * frame extents with _NET_REQUEST_FRAME_EXTENTS
```
s/with _NET/via _NET/

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Screen.c:
```diff
@@ -314,6 +314,28 @@ void nxagentMinimizeFromFullScreen(ScreenPtr pScreen)
   }
 }
 
+/* check if possible_parent id parent of candidate */
```
s/id/is/

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Screen.c:
```diff
@@ -336,6 +336,11 @@ Bool nxagentIsParentOf(Display *d, Window possible_parent, Window candidate)
 
+/*
+ * In contrast to its name this function is here to take care of the
+ * iconfied window. Some window managers tend to do 'interesting'
```
s/iconfied/iconified/ (which window do you refer to here? The nxagent window? Please explain better)

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Screen.c:
```diff
@@ -349,44 +354,70 @@ void nxagentMaximizeToFullScreen(ScreenPtr pScreen)
-    XReparentWindow(nxagentDisplay, nxagentFullscreenWindow,
-                        RootWindow(nxagentDisplay, DefaultScreen(nxagentDisplay)), 0, 0);
+    /* only reparent if necessary. FIXME: also check if the desired coordinates match*/
 
```
s@match*/@match */@ (blank missing near EOL)

sunweaver commented on nx-X11/programs/Xserver/hw/nxagent/Screen.c:
```diff
@@ -349,44 +354,70 @@ void nxagentMaximizeToFullScreen(ScreenPtr pScreen)
+    else
+    {
+      #ifdef TEST
+      fprintf(stderr, "%s: FullscreenWindow already has root has parent - skipping reparenting,\n", __func__);
```
s/has root has parent/has root as parent/

uli42 commented:
Amended. And added some more little changes. Please review, especially the explanations.

I am not 100% sure if my patch is really fixing all the problems. It works for me, but it might nor work for others. So I was a bit hesitant. Changed it...
